htdocs/core/class/html.form.class.php

```diff
@@ -3553,9 +3553,9 @@ public function select_produits_fournisseurs_list($socid, $selected = '', $htmln
 						'value'=>$outref,
 						'label'=>$outval,
 						'qty'=>$outqty,
-						'price_qty_ht'=>price2num($objp->fprice, 'MU'),	// Keep higher resolution for price for the min qty
-						'price_unit_ht'=>price2num($objp->unitprice, 'MU'),	// This is used to fill the Unit Price
-						'price_ht'=>price2num($objp->unitprice, 'MU'),		// This is used to fill the Unit Price (for compatibility)
+						'price_qty_ht'=>price($objp->fprice),		// Keep higher resolution for price for the min qty -> local format settings need to be considered
```

Comment:
What if you use   
`price(price2num($objp->fprice, 'MU'))`
So we keep the rounding on MU = Unit Amount and we return data in local string format.
Is this will be ok for you ?